iidm/iidm-criteria/src/main/java/com/powsybl/iidm/criteria/translation/DefaultNetworkElementAdapter.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.criteria.translation;
+
+import com.powsybl.iidm.criteria.NetworkElementCriterion.NetworkElementCriterionType;
+import com.powsybl.iidm.network.*;
+
+import java.util.Optional;
+import java.util.Set;
+
+public class DefaultNetworkElementAdapter implements NetworkElement {
+
+    private static final Set<IdentifiableType> VALID_TYPES_FOR_LINE_CRITERION = Set.of(IdentifiableType.LINE, IdentifiableType.TIE_LINE);
+
+    private final Identifiable<?> identifiable;
+
+    public DefaultNetworkElementAdapter(Identifiable<?> identifiable) {
+        this.identifiable = identifiable;
+    }
+
+    @Override
+    public String getId() {
+        return identifiable.getId();
+    }
+
+    @Override
+    public Country getCountry1() {
+        return getCountry(TwoSides.ONE);
+    }
+
+    @Override
+    public Country getCountry2() {
+        return getCountry(TwoSides.TWO);
+    }
+
+    @Override
+    public Country getCountry() {
+        switch (identifiable.getType()) {
+            case TWO_WINDINGS_TRANSFORMER -> {
+                Optional<Substation> substation = ((TwoWindingsTransformer) identifiable).getSubstation();
+                return substation.map(Substation::getNullableCountry).orElse(null);
+            }
+            case THREE_WINDINGS_TRANSFORMER -> {
+                Optional<Substation> substation = ((ThreeWindingsTransformer) identifiable).getSubstation();
+                return substation.map(Substation::getNullableCountry).orElse(null);
+            }
+            default -> {
+                Country country1 = getCountry1();
+                return country1 != null ? country1 : getCountry2();
+            }
```

Comment:
for a line it returns the country of the side one ? if the line is between two countries is it good ?